types/contentTypes.ts

```diff
@@ -38,6 +38,8 @@ export type CommentAuthorPartial = {
   authorUser?: StrapiUser;
 };
 
+export type CommentAuthorResolved = CommentAuthor & StringMap<unknown>;
```

Comment:
```suggestion
export type CommentAuthorResolved<TExtension = StringMap<unknown>> = CommentAuthor & TExtension;
```
Not sure for this specific type usage, but I think that using generic type flexibility when needed and escape hatch that, I have an impression, is needed.